siege_utilities/reporting/client_branding.py

```
@@ -12,6 +13,35 @@
+
+    Previously the branding manager wrote to a directory whose name was
+    just `client_name.lower().replace(" ", "_")`. That left every other
+    piece of punctuation untouched — including `../` and OS path
+    separators — so a hostile client_name could escape config_dir.
+    The new normaliser is an allow-list: lowercase, spaces → `_`,
+    everything else stripped. Two inputs that slugify to the same value
+    collide; that's the same behaviour as before for spaces, just now
+    consistent for all punctuation.
+    """
+    if not isinstance(client_name, str) or not client_name.strip():
+        raise ValueError(f"client_name must be a non-empty string, got {client_name!r}")
```

Ruff (0.15.12)

[warning] Avoid specifying long messages outside the exception class

(TRY003)